crates/alloy-celo-evm/src/block/receipt_builder.rs

```diff
@@ -22,6 +36,44 @@ impl OpReceiptBuilder for CeloAlloyReceiptBuilder {
         ctx: ReceiptBuilderCtx<'a, CeloTxEnvelope, E>,
     ) -> Result<Self::Receipt, ReceiptBuilderCtx<'a, CeloTxEnvelope, E>> {
         match ctx.tx.tx_type() {
+            CeloTxType::Cip64 => {
+                let base_fee = ctx.evm.block().basefee as u128;
+                // For CIP-64 transactions, calculate the base fee in ERC20
+                let base_fee_in_erc20 = if let CeloTxEnvelope::Cip64(cip64) = ctx.tx {
+                    if let Some(fee_currency) = cip64.tx().fee_currency {
+                        // Try to get context from instance first, then fall back to global
+                        if let Some(ctx) = &self.fee_currency_context {
+                            ctx.celo_to_currency(Some(fee_currency), U256::from(base_fee))
+                                .ok()
+                                .and_then(|v| v.try_into().ok())
+                        } else if let Some(global_ctx) =
+                            celo_revm::global_context::get_fee_currency_context()
+                        {
+                            global_ctx
+                                .celo_to_currency(Some(fee_currency), U256::from(base_fee))
+                                .ok()
+                                .and_then(|v| v.try_into().ok())
+                        } else {
+                            // If no context available, just use the base fee
+                            Some(base_fee)
+                        }
+                    } else {
+                        Some(base_fee)
```

Comment:
I believe this code is unreachable, but wouldn't it be reasonable to set it to `None` if `fee_currency` is `None`?
```suggestion
                        None
```